core/aws_ddk_core/stages/glue_transform.py

```diff
@@ -108,6 +108,7 @@ def __init__(
             self._crawler = CfnCrawler(
                 self,
                 f"{id}-crawler",
+                name=f"{id}_crawler",
```

Comment:
Hey, thanks for point out this issue.

Generally, hardcoding resource names in CDK is considered bad practice. I think a better solution would be to let CDK generate the name, and then reference the name in like 138 with:
```python
"Parameters": {"Name": self._crawler.name},
```

Alternatively, the [ref property](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-glue-crawler.html#aws-resource-glue-crawler-return-values) on the Crawler resource will return the crawler name.

Reply:
I tested the behaviour with the following


```
#!/usr/bin/env python3

import aws_cdk as cdk
from aws_cdk import aws_glue as glue


app = cdk.App()
stack = cdk.Stack(app, 'this')


# tags: Any
cfn_crawler_without_name = glue.CfnCrawler(stack, "MyCfnCrawler1",
    role="role",
    targets=glue.CfnCrawler.TargetsProperty(
        s3_targets=[glue.CfnCrawler.S3TargetProperty(
            connection_name="connectionName",
            dlq_event_queue_arn="dlqEventQueueArn",
            event_queue_arn="eventQueueArn",
            exclusions=["exclusions"],
            path="path",
            sample_size=123
        )]
    ),
)
cfn_crawler_with_name = glue.CfnCrawler(stack, "MyCfnCrawler2",
    name="ThisOneHasAName",
    role="role",
    targets=glue.CfnCrawler.TargetsProperty(
        s3_targets=[glue.CfnCrawler.S3TargetProperty(
            connection_name="connectionName",
            dlq_event_queue_arn="dlqEventQueueArn",
            event_queue_arn="eventQueueArn",
            exclusions=["exclusions"],
            path="path",
            sample_size=123
        )]
    ),
)
print(f"Crawler Name without `name` set: .name=={cfn_crawler_without_name.name}")
print(f"Crawler Name with `name` set: .name=={cfn_crawler_with_name.name}")

app.synth()
```


output: 

```
> cdk synth
Crawler Name without `name` set: .name==None
Crawler Name with `name` set: .name==ThisOneHasAName
```

Reply:
What about doing _crawler.ref?

Reply:
Gives us a `${Token[TOKEN.206]}` possibly that can be resolved in the SFN statemachine correctly

Reply:
Yap I tried to run the following:
```python
import aws_cdk as cdk
from aws_cdk import aws_glue as glue
from constructs import Construct

class TestStack(cdk.Stack):

    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        cfn_crawler = glue.CfnCrawler(self, "My Crawler",
            role="role",
            targets=glue.CfnCrawler.TargetsProperty(
                s3_targets=[glue.CfnCrawler.S3TargetProperty(
                    connection_name="connectionName",
                    dlq_event_queue_arn="dlqEventQueueArn",
                    event_queue_arn="eventQueueArn",
                    exclusions=["exclusions"],
                    path="path",
                    sample_size=123
                )]
            ),
        )

        cdk.CfnOutput(self, "Crawler Name", value=cfn_crawler.ref)
```

When I did `cdk synth` on this stack, the output of the stack ended up being `Ref: MyCrawler`, which will resolve to the crawler name.

Reply:
Nice, we should proceed with that then, thanks @LeonLuttenberger